src/libcrun/cgroup.c

```diff
@@ -2828,6 +2828,26 @@ update_cgroup_v2_resources (runtime_spec_schema_config_linux_resources *resource
         return ret;
     }
 
+  /* Write unified resources if any.  They have higher precedence and override any previous setting.  */
+  if (resources->unified)
+    {
+      size_t i;
+
+      for (i = 0; i < resources->unified->len; i++)
+        {
+          size_t len;
+
+          if (strchr (resources->unified->keys[i], '/'))
+            return crun_make_error (err, 0, "key `%s` must be a file name without any slash", resources->unified->keys[i]);
+
+          len = strlen (resources->unified->values[i]);
+          ret = write_file_at (cgroup_dirfd, resources->unified->keys[i], resources->unified->values[i], len, err);
```

Comment:
How do we handle validation for the values?

Reply:
We could punt to the kernel entirely but we should ensure we are returning a clear error that we failed writing to unified cgroup value and advise user to check the kernel documentation for the accepted values for a setting.

Reply:
agree, I've added some logic to return more useful error messages when the controller is not present